src/status_im/contexts/wallet/add_account/create_account/events.cljs

```diff
@@ -202,3 +203,38 @@
                                    derived-from-address
                                    %)}]]]}))
 
+(rf/reg-event-fx
+ :wallet/set-seed-phrase
+ (fn [{:keys [db]} [{:keys [seed-phrase on-success]}]]
+   {:db (assoc-in db [:wallet :ui :create-account :new-keypair :seed-phrase] seed-phrase)
+    :fx [on-success]}))
+
+(defn import-private-key-and-create-keypair-with-account
+  [{:keys [db]} [{:keys [password account-preferences]}]]
+  (let [private-key          (get-in db [:wallet :ui :create-account :private-key])
+        unmasked-private-key (security/safe-unmask-data private-key)
+        unmasked-password    (security/safe-unmask-data password)]
+    {:json-rpc/call
+     [{:method     "accounts_importPrivateKey"
+       :params     [unmasked-private-key unmasked-password]
+       :on-success [:wallet/import-and-create-keypair-with-account
+                    {:passord             password
+                     :account-preferences account-preferences}]
+       :on-error   #(log/error "Failed to import private key" %)}]}))
+
+(rf/reg-event-fx
+ :wallet/import-private-key-and-create-keypair-with-account
+ import-private-key-and-create-keypair-with-account)
+
+(rf/reg-event-fx
+ :wallet/import-private-key-and-generate-account-for-keypair
+ (fn [_ [{:keys [keypair-name]}]]
+   {:effects.wallet/get-random-mnemonic
+    {:on-success
+     #(rf/dispatch
+       [:wallet/set-seed-phrase
+        {:seed-phrase %
+         :on-success  [:dispatch
```

Comment:
I see no reason for this dispatch to be an argument of `set-seed-phrase`, but please correct me if I'm missing something.

Reply:
Yeah, I think this and the `set-seed-phrase` event need to be refactored, the logic seems a bit complex :thinking:

Reply:
`set-seed-phrase` event removed